Wrappers/Python/cil/optimisation/functions/MixedL21Norm.py

```diff
@@ -83,48 +143,25 @@ def proximal(self, x, tau, out=None):
         where the convention 0 · (0/0) = 0 is used.
         
         """
-
-        # Note: we divide x/tau so the cases of both scalar and 
-        # datacontainers of tau to be able to run
-        
+                
+        tmp = x.pnorm(2)
+        if has_numba and isinstance(tau, Number):
+            resarray = _proximal_step_numba(np.asarray(tmp.as_array(), order='C', dtype=np.float32), 
```

Comment:
The asarray casting type should also be explicitly set, I think we want to use 'safe' to keep the precision the user set.

Reply:
But there is no option for casting method in [`asarray`](https://numpy.org/doc/stable/reference/generated/numpy.asarray.html) as in [`astype`](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.astype.html).

Reply:
ah, that's more of an issue. what about only using the numba implementation if it's float32 already then? We need a safe route for complex data at least.